automation/taskcluster/decision_task.py

```diff
@@ -3,272 +3,144 @@
-            "source": "https://github.com/mozilla-mobile/android-components"
-        }
-    }
-
-
-def create_variant_assemble_task(variant):
-    return create_task(
-        name="assemble: %s" % variant,
-        description='Building and testing variant ' + variant,
-        command='assemble{} && ls -R /build/reference-browser/'.format(variant.capitalize()),
-        treeherder={
-            'jobKind': 'build',
-            'machine': {
-              'platform': _craft_treeherder_platform_from_variant(variant),
-            },
-            'symbol': 'A',
-            'tier': 1,
-        },
-        artifacts=_craft_artifacts_from_variant(variant),
-    )
-
+def pr_or_push(is_push=False):
+    if SKIP_TASKS_TRIGGER in PR_TITLE:
+        print("Pull request title contains", SKIP_TASKS_TRIGGER)
+        print("Exit")
+        return {}
 
-def create_variant_test_task(variant):
-    return create_task(
-        name="test: %s" % variant,
-        description='Building and testing variant ' + variant,
-        command='test{}UnitTest && ls -R /build/reference-browser/'.format(variant.capitalize()),
-        treeherder={
-            'jobKind': 'test',
-            'machine': {
-              'platform': _craft_treeherder_platform_from_variant(variant),
-            },
-            'symbol': 'T',
-            'tier': 1,
-        },
-    )
+    print("Fetching build variants from gradle")
+    variants = build_variants.from_gradle()
 
-def _craft_treeherder_platform_from_variant(variant):
-    architecture, build_type = _get_architecture_and_build_type_from_variant(variant)
-    return 'android-{}-{}'.format(architecture, build_type)
+    if len(variants) == 0:
+        raise ValueError("Could not get build variants from gradle")
 
+    print("Got variants: {}".format(' '.join(variants)))
 
-def _craft_artifacts_from_variant(variant):
-    arch, _ = _get_architecture_and_build_type_from_variant(variant)
-    return {
-        'public/target.{}.apk'.format(arch): {
-            'type': 'file',
-            'path': _craft_apk_full_path_from_variant(variant),
-            'expires': taskcluster.stringDate(taskcluster.fromNow(lib.tasks.DEFAULT_EXPIRES_IN)),
-        }
-    }
+    build_tasks = {}
+    signing_tasks = {}
+    other_tasks = {}
 
+    for variant in variants:
+        assemble_task_id = taskcluster.slugId()
+        build_tasks[assemble_task_id] = BUILDER.craft_assemble_task(variant)
+        architecture, build_type, _ = get_architecture_and_build_type_and_product_from_variant(
+            variant
+        )
 
-def _craft_apk_full_path_from_variant(variant):
-    architecture, build_type = _get_architecture_and_build_type_from_variant(variant)
+        if (
+            is_push and
+            architecture in ('aarch64', 'arm') and
+            build_type == 'releaseraptor'
+        ):
+            signing_tasks[taskcluster.slugId()] = BUILDER.craft_signing_task(assemble_task_id, variant)
```

Comment:
I'm sorry, I'm not sure to follow. This line enables dep-signin on pushes outside of PRs. I note this occurs on all branches, which we don't need at the moment. I can restrict that to master, if needed. Please let me know what you think!

Reply:
Ohhhh, PRs don't show up as branches, right!
I'd prefer if we can restrict this to master, so that if we have other behaviour happen on-master-push, we don't accidentally have it triggered when a currently-unknown branch springs up in the base repository :)

Reply:
Okay. Fixed!